src/components/calculator.js

```diff
@@ -1,136 +1,190 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor() {
     super();
-    this.state = {};
+    this.state = {
+      calculatorObject: {
+        total: null,
+        next: null,
+        operation: null,
+      },
+    };
   }
 
+  clicked = (e) => {
+    const result = calculate(this.state, e.target.textContent);
+    this.setState(result);
+  };
+
   render() {
+    const { total, next } = this.state;
     return (
       <fragment>
```

Comment:
- [ ] [OPTIONAL] I got this error in the console `react-dom.development.js:86 Warning: The tag <fragment> is unrecognized in this browser, kindly consider using an empty tag `<>.....</>` to use fragment

Reply:
Done! Thanks so much.